### PR TITLE
add(decompression service)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Adds decompression service part of https://github.com/Clinical-Genomics/add-new-tech/issues/100

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/decompression_service/decompressor.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from pathlib import Path
+
+
+class Decompressor:
+
+    def decompress(self, source_path: Path, destination_path: Path) -> Path:
+        """Decompress a file to a directory."""
+        with zipfile.ZipFile(source_path, "r") as zip_file:
+            zip_file.extractall(destination_path)
+        if not self._decompressed_path_exists(destination_path):
```
Comment (islean) [inline cg/services/decompression_service/decompressor.py:11]: A bit uncertain of the timings of it all, but is the decompression async?

Comment (islean) [reply cg/services/decompression_service/decompressor.py:11]: Or could the decompression be running here still?

**Review by islean — APPROVED** [review]
Looks good! Is it the intent to at some point merge the CompressAPI and the decompression service? Seems to be some overlap with the intent?

**Comment by ChrOertlin** [comment]
> Looks good! Is it the intent to at some point merge the CompressAPI and the decompression service? Seems to be some overlap with the intent?

Perhaps, I don't know. The compress and decompress api are tihgtly linked to flow cells and fastq files. We can have a look at that when we get to that code. This here is, to me, considered apart from that.

File: cg/services/decompression_service/decompressor.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/services/decompression_service/decompressor.py:11]: As far as I understand it just unzips the folder, almost instant?

**Comment by ChrOertlin** [comment]
ignore coveralls

**Pull request merged**